prow/cmd/webhook-server/clients.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/test-infra/experiment/clustersecretbackup/secretmanager"
+)
+
+type GCPClient struct {
+	client *secretmanager.Client
+}
+
+func newGCPClient(client *secretmanager.Client) *GCPClient {
+	return &GCPClient{
+		client: client,
+	}
+}
+
+func (g *GCPClient) CreateSecret(ctx context.Context, secretID string) error {
+	_, err := g.client.CreateSecret(ctx, secretID)
+	if err != nil {
+		return fmt.Errorf("could not create secret %v", err)
+	}
+	return nil
+}
+
+func (g *GCPClient) AddSecretVersion(ctx context.Context, secretName string, payload []byte) error {
+	if err := g.client.AddSecretVersion(ctx, secretName, payload); err != nil {
+		return fmt.Errorf("could not add secret data %v", err)
+	}
+	return nil
+}
+
+func (g *GCPClient) GetSecretValue(ctx context.Context, secretName string, versionName string) ([]byte, error) {
```

Comment:
We should only call CheckSecret once, either in this function or outside it. I think we should do it in this function and change the signature to also return a bool indicating if the secret was found.